app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -0,0 +1,315 @@
+  /// pull request.
+  Future<String> _getGoldStatus(PullRequest pr, Logging log) async {
+    // We wait for a few seconds in case tests _just_ finished and the tryjob
+    // has not finished ingesting the results.
+    await Future<void>.delayed(const Duration(seconds: 10));
+    final Uri requestForTryjobStatus = Uri.parse(
+        'http://flutter-gold.skia.org/json/changelist/github/${pr.number}/${pr.head.sha}/untriaged');
+    String rawResponse;
+    try {
+      final HttpClientRequest request =
+          await goldClient.getUrl(requestForTryjobStatus);
+      final HttpClientResponse response = await request.close();
+
+      rawResponse = await utf8.decodeStream(response);
+      final Map<String, dynamic> decodedResponse = json.decode(rawResponse);
+
+      if (decodedResponse['digests'] == null) {
+        log.debug(
+            'There are no unexpected image results for ${pr.number} at sha '
+            '${pr.head.sha}, returning status ${GithubGoldStatusUpdate.statusCompleted}');
+
+        return GithubGoldStatusUpdate.statusCompleted;
+      } else {
+        log.debug('Tryjob for ${pr.number} at sha ${pr.head.sha} generated new '
+            'images, returning status ${GithubGoldStatusUpdate.statusRunning}');
+
+        return GithubGoldStatusUpdate.statusRunning;
+      }
+    } on FormatException catch (_) {
+      throw BadRequestException('Formatting error detected requesting '
+          'tryjob status for pr ${pr.number} from Flutter Gold.\n'
+          'rawResponse: $rawResponse');
+    } catch (e) {
+      throw BadRequestException(
+          'Error detected requesting tryjob status for pr '
+          '${pr.number} from Flutter Gold.\n'
+          'error: $e');
+    }
+  }
+
+  String _getStatusDescription(String status) {
+    if (status == GithubGoldStatusUpdate.statusRunning)
+      return 'Image changes have been found for '
+          'this pr. Visit https://flutter-gold.skia.org/changelists to '
+          'view and triage (e.g. because this is an intentional change).';
+    return 'All golden file tests have passed.';
+  }
+
+  /// Creates a comment on a given pull request identified to have golden file
+  /// changes and applies the `will affect goldens` label.
+  Future<void> _commentAndApplyGoldLabel(
+    GitHub gitHubClient,
+    PullRequest pr,
+    RepositorySlug slug,
+  ) async {
+    final String body = 'Golden image changes have been found for this pull '
+            'request. Click [here](https://flutter-gold.skia.org/search?issue=${pr.number}&new_clstore=true) '
+            'to view and triage (e.g. because this is an intentional change).\n\n' +
+        config.goldenBreakingChangeMessage +
+        '\n\n' +
+        '_Changes reported for pull request ${pr.number} at sha ${pr.head.sha}_\n\n';
+    await gitHubClient.issues.createComment(slug, pr.number, body);
+    await gitHubClient.issues.addLabelsToIssue(slug, pr.number, <String>[
+      'will affect goldens',
+      'severe: API break',
```

Comment:
Should also add the `CQ+1` label to trigger a Luci build and update those goldens. Currently, Luci builds are only triggered on-demand in pre-submit. If were are going to support Luci goldens (https://github.com/flutter/flutter/pull/49815), we will need to make sure they get updated along with the rest when changes are made.

Reply:
Added a to-do as Luci goldens are not yet enabled.